JEasyCryptoConsole/src/EasyCryptoConsole/EasyCryptoConsole.java

```diff
@@ -6,58 +6,59 @@
+                    EasyCryptoAPI.Result result;
 
-	public static void main(String[] args) {
-
-		try {
-			System.setOut(new PrintStream(System.out, true, "UTF-8"));
-			System.setProperty("file.encoding", "UTF-8");
-			Console console = System.console();
-
-			console.printf("Welcome to CryptoClient!\n");
-			console.printf("Supported methods are: %s\n", EasyCryptoAPI.methods());
-
-			while (true) {
-				String d = console.readLine("Do you wish to encrypt or decrypt (e or d)? > ");
-				
-
-				EasyCryptoAPI.Result result;
-
-				if (d.equalsIgnoreCase("e")) {
-					String e = console.readLine("Please enter text to be encrypted > ");
-					String m = console.readLine("Please enter encryption method > ");
-					result = EasyCryptoAPI.encrypt(e, m);
-				} else if (d.equalsIgnoreCase("d")) {
-					String e = console.readLine("Please enter text to be decrypted > ");
-					String m = console.readLine("Please enter decryption method > ");
-					result = EasyCryptoAPI.decrypt(e, m);
-				} else {
-					return;
-				}
-		        
-				console.printf("\nResult is: %d - %s\n", result.resultCode().ordinal(), result.resultCode().toString());
-				switch (result.resultCode()) {
-				case ESuccess: {
-					console.printf("Encrypted text is: %s \n", result.result());
-					System.out.println(result.result());
-					break;
-				}
-				case EError:
-				case ENotSupported: {
-					console.printf("Encrypted text is: %s \n", result.result());
-					break;
-				}
-				}
-			}
-
-		} catch (UnsupportedEncodingException e1) {
-			e1.printStackTrace();
-		} catch (Exception e) {
-			e.printStackTrace();
-		}
-	}
+                    if (d.equalsIgnoreCase("e") || d.equalsIgnoreCase("d") || d.equalsIgnoreCase("q")) {
+                        if (d.equalsIgnoreCase("e")) {
+                            String e = console.readLine("Please enter text to be encrypted > ");
+                            String m = console.readLine("Please enter encryption method > ");
+                            result = EasyCryptoAPI.encrypt(e, m);
+                        } else if (d.equalsIgnoreCase("d")) {
+                            String e = console.readLine("Please enter text to be decrypted > ");
+                            String m = console.readLine("Please enter decryption method > ");
+                            result = EasyCryptoAPI.decrypt(e, m);
+                        } else {
+                            return;
+                        }
 
+                        console.printf("\nResult is: %d - %s\n", result.resultCode().ordinal(), result.resultCode().toString());
+                        switch (result.resultCode()) {
+                            case ESuccess: {
+                                console.printf("Encrypted text is: %s \n", result.result());
+                                System.out.println(result.result());
+                                break;
+                            }
+                            case EError:
```

Comment:
What about the `case EError`? We should handle that somehow!

Reply:
This ```case EError:``` isn't exactly in the scope of this pull request as the focus is specifically in method selection. ```case EError:``` only handles problems happening during the encrypting/decrypting text and in addition it already seems to be handled just like ```case ENotSupported: ```. I would suggest that this shouldn't be touched at this point.

Reply:
True. It is definitely not in the scope of the PR. It is however bad design. But I agree with you, that it has to be fixed somewhere else!